apps/src/gamelab/GameLabSprite.js

```diff
@@ -42,210 +41,220 @@ module.exports.createSprite = function (x, y, width, height) {
-    return s.animation ? s.animation.frameChanged : false;
-  };
-
-  s.pointTo = function (x, y) {
-    var yDelta = y - s.position.y;
-    var xDelta = x - s.position.x;
-    if (!isNaN(xDelta) && !isNaN(yDelta) && (xDelta !== 0 || yDelta !== 0)) {
-      var radiansAngle = Math.atan2(yDelta, xDelta);
-      s.rotation = 360 * radiansAngle / (2 * Math.PI);
-    }
-  };
-
-  // The scale value should include the horizontal stretch for animations.
-  s._getScaleX = function () {
-    return s.scale * s._horizontalStretch;
-  };
-
-  // The scale value should include the vertical stretch for animations.
-  s._getScaleY = function () {
-    return s.scale * s._verticalStretch;
-  };
-
-  /*
-   * @private
-   * For game lab, don't update the animation sizes because all frames are the same size.
-   */
-  s._syncAnimationSizes = function (animations, currentAnimation) {};
-
-  Object.defineProperty(s, 'frameDelay', {
-    enumerable: true,
-    get: function () {
-      if (s.animation) {
-        return s.animation.frameDelay;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s.animation.frameDelay = value;
-      }
-    }
-  });
-
-  // Overriding these allows users to set a width for
+  // Overriding these allows users to set a width and height for
   // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'width', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalWidth === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalWidth * s._horizontalStretch;
-      } else {
-        return s._internalWidth;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._horizontalStretch = value / s._internalWidth;
-      } else {
-        s._internalWidth = value;
-      }
-    }
+    get: getWidth,
+    set: setWidth
   });
-
-  // Overriding these allows users to set a height for
-  // an animated sprite the same way they would an unanimated sprite.
   Object.defineProperty(s, 'height', {
     enumerable: true,
     configurable: true,
-    get: function () {
-      if (s._internalHeight === undefined) {
-        return 100;
-      } else if (s.animation) {
-        return s._internalHeight * s._verticalStretch;
-      } else {
-        return s._internalHeight;
-      }
-    },
-    set: function (value) {
-      if (s.animation) {
-        s._verticalStretch = value / s._internalHeight;
-      } else {
-        s._internalHeight =  value;
-      }
-    }
+    get: getHeight,
+    set: setHeight
   });
 
-  // p5.play stores width unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledWidth = function () {
-    return s.width * s.scale;
-  };
-
-  // p5.play stores height unscaled, but users in
-  // Game Lab should have access to a scaled version.
-  s.getScaledHeight = function () {
-    return s.height * s.scale;
-  };
```

Comment:
There's a lot to take in here, but the general pattern is that for each of these functions we've added or overridden on the p5.play sprite, we were creating a new `Function` object for every single sprite instead of having one function that all sprites could refer to.  I've remedied this by moving the function implementations into the module scope (see below) and just assigning them to the created sprite here.  This has the added benefit that there's a little less scope magic going on there, and it's easier to read the `createSprite` method from start to end.